### PR TITLE
`azurerm_security_center_subscription_pricing`: Upgrade API, add `subplan` argument

### LINKED ISSUE #19004: Support for Defender CSPM for Defender for cloud plans
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Microsoft are actively developing the Defender for Cloud. 

Current TF code does not enable 'Defender CSPM' free plan at the moment. Before it was 'Cloud Security Posture management free' and it would be enabled by default as soon as any advanced plans are enabled.

This is request to add an additional plan to azurerm_security_center_subscription_pricing resource to control this basic plan.

### New or Affected Resource(s)/Data Source(s)

azurerm_azurerm_security_center_subscription_pricing

### Potential Terraform Configuration

_No response_
Cli implementation 
```
az security pricing list

az security pricing create -n CloudPosture --tier 'standard'
```
It seems that we just need to allow CloudPosture.

I also think CosmosDB plan is not available.

### References

_No response_

### PR DESCRIPTION
**Pull request by favoretti**
Also added `CloudPosture` as allowed value.
Fixes #19004

```
$ TF_ACC=1 go test -v ./internal/services/securitycenter -timeout=1000m -run='TestAccSecurityCenterSubscriptionPricing'
=== RUN   TestAccSecurityCenterSubscriptionPricing_update
--- PASS: TestAccSecurityCenterSubscriptionPricing_update (174.34s)
=== RUN   TestAccSecurityCenterSubscriptionPricing_storageAccountSubplan
--- PASS: TestAccSecurityCenterSubscriptionPricing_storageAccountSubplan (90.04s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/securitycenter	265.614s
```

### CODE REVIEW

**Comment by favoretti** [comment]
image

File: internal/services/securitycenter/security_center_subscription_pricing_resource.go [hunk 1]
```diff
@@ -115,9 +126,10 @@ func resourceSecurityCenterSubscriptionPricingRead(d *pluginsdk.ResourceData, me
 		return fmt.Errorf("retrieving %s: %+v", *id, err)
 	}
 
-	d.Set("resource_type", id.Name)
-	if properties := resp.PricingProperties; properties != nil {
+	d.Set("resource_type", id.PricingName)
+	if properties := resp.Model.Properties; properties != nil {
```
Comment (stephybun) [inline internal/services/securitycenter/security_center_subscription_pricing_resource.go:130]: Can we nil check Model here as well?

**Review by stephybun — COMMENTED** [review]
Thanks @favoretti, one minor comment otherwise this looks good 👍

File: internal/services/securitycenter/security_center_subscription_pricing_resource.go (hunk 1 above)
Comment (favoretti) [reply internal/services/securitycenter/security_center_subscription_pricing_resource.go]: Certainly, pushed a fix.

**Review by stephybun — APPROVED** [review]
Thanks @favoretti LGTM 💯

**Pull request merged**